### ISSUE TITLE
Wrong dynamic aliases generating in 1.9.3 (in 1.9.1 that works fine!)

### ISSUE THREAD

**Issue by Beszt** [doxygen/doxygen#9052]
**Describe the bug**
When i try generated new "Related pages" with **xrefitem** and using a dynamic variables in *ALIASES* my output for *first* link in generated list was good, but other links it's going out of table's bounds! 

**To Reproduce**
Example code in config file:
````
ALIASES = "sw_requirement{1}=\xrefitem sw_req_index \"Index of software requirement\" \"Index of software requirements\" <a href=\"sw_requirements.html#\1\">\1</a>" \
          "test_requirement{1}=\xrefitem test_req_index \"Index of test requirement\" \"Index of test requirements\" <a href=\"test_requirements.html#\1\">\1</a>"
````
Example code in example C# file:
```
/// <summary>
/// DUT diagnosic client (send request, wait for response, manage diagnostic session)<br/>
/// @sw_requirement{SW-DSPL-IPC_2}
/// </summary>
```
**Version**
Thats bug occurs in 1.9.3, in 1.9.1 that feature works good!

**Comment by albert-github** [comment]
In your small example there is only one link and all looks OK to me, might be that I missed something, so:
- Can you please attach a, small, self contained example (source+configuration file in a tar or zip) that allows us to reproduce the problem? Please don't add external links as they might not be persistent.

**Comment by albert-github** [comment]
Revisiting and paying some attention to "but other links it's going out of table's bounds!", so creating a small example with 2 times using `but other links it's going out of table's bounds!` like:
**aa.cs**
```
/// \file
///
/// <summary>
/// DUT diagnosic client (send request, wait for response, manage diagnostic session)<br/>
/// @sw_requirement{SW-DSPL-IPC_aa}
/// </summary>
```
**bb.cs**
```
/// \file
///
/// <summary>
/// DUT_bb diagnosic client (send request, wait for response, manage diagnostic session)<br/>
/// @sw_requirement{SW-DSPL-IPC_bb}
/// </summary>
```
With the 1.9.3 (and 1.9.4) version we get the warning:
```
sw_req_index:4: warning: Unexpected tag <dd> found
```
and the output looks like:

![image](https://user-images.githubusercontent.com/5223533/179352593-efce010d-77e7-4ec9-ad1f-f43ace388750.png)

With  the 1.9.1 and the current master (1.9.5 (3b30f16032a4b4512c2b60fb4b1cf68c36d06854)) version the warning is not present and the output looks like:

![image](https://user-images.githubusercontent.com/5223533/179352642-4bd3bc5b-2cec-43ab-b699-e2f5feaa458c.png)


So this looks all OK to me.

I think the problem is fixed but not released in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.5.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
